plugins/extractors/maxcompute/client.go

```diff
@@ -0,0 +1,72 @@
+package maxcompute
+
+import (
+	"context"
+
+	"github.com/aliyun/aliyun-odps-go-sdk/odps"
+	"github.com/aliyun/aliyun-odps-go-sdk/odps/account"
+	"github.com/aliyun/aliyun-odps-go-sdk/odps/tableschema"
+	"github.com/goto/meteor/plugins"
+	"github.com/goto/salt/log"
+)
+
+type MCClient struct {
+	client  *odps.Odps
+	project *odps.Project
+	log     log.Logger
+}
+
+func NewMaxComputeClient(config Config) *MCClient {
+	aliAccount := account.NewAliyunAccount(config.AccessKey.ID, config.AccessKey.Secret)
+	client := odps.NewOdps(aliAccount, config.EndpointProject)
+	client.SetDefaultProjectName(config.ProjectName)
+
+	project := client.Project(config.ProjectName)
+
+	return &MCClient{
+		client:  client,
+		project: project,
+		log:     plugins.GetLog(),
+	}
+}
+
+func (c *MCClient) ListSchema(context.Context) (schemas []*odps.Schema, err error) {
+	err = c.project.Schemas().List(func(schema *odps.Schema, err2 error) {
```

Comment:
just realized that schemas is a feature that not always enabled, we can check first if the feature are enabled or not
https://www.alibabacloud.com/help/en/maxcompute/user-guide/schema-related-operations?spm=a2c63.p38356.0.0.43c85b491WqcRQ

```
...
type MCClient struct {
       ...
	isSchemaEnabled true
}
...

func NewMaxComputeClient(config Config) (*MCClient, error) {
  ...
  properties, err := project.GetAllProperties()
  isSchemaEnabled :=  properties.Get("odps.tableschema.enabled") == "true"
  ...
}
...
func (c *MCClient) ListSchema(context.Context) (schemas []*odps.Schema, err error) {
  ...
  // return default schemas if not enabled
  ...
}